### PR TITLE
Modernize repo, fix types, a few bugfixes

### PR DESCRIPTION
**Pull request by bradymadden97**
Types are messed up, most of the types are not returning promises, which can cause bugs when trying to use this library. See https://replit.slack.com/archives/C030RJ2PRC7/p1711850728380709

Took the opportunity to try and modernize the repo, instead of manually fixing the types, we should generate the types file from the src so it's accurate.

* Rewrote the library in typescript
* Rewrote the tests in typescript
* Generate the library using `unbuild`
* Fixed a bug in `delete` when deleting keys with `\n`
* Fixed a bug that kept `setInterval` open indefinitely which prevented processing using this library from ever exiting

Tests pass locally:

Screen Shot 2024-03-30 at 11 25 23 PM

Generated types file in `/dist` looks good:

image

### CODE REVIEW

**Comment by mikewesthad** [comment]
:pog:

**Review by masad-frost — APPROVED** [review]
:clap:

File: src/index.ts [hunk 1]
```diff
@@ -60,7 +78,7 @@ class Client {
    * @param {String} key Key
    * @param {any} value Value
    */
-  async set(key, value) {
+  async set<Value>(key: string, value: Value) {
```
[the remark this answers is not in the document]
Comment (bradymadden97) [reply src/index.ts:81]: didn't really know what to put here, should i just use any?

**Comment by bradymadden97** [comment]
What bugs are you referring to? :)

**Pull request merged**